### PR TITLE
adds preauth sample and postauth method

### PR DESCRIPTION
**Pull request by AlicanAkkus**

### CODE REVIEW

File: src/Model/PaymentPhase.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Craftgate\Model;
+
+class PaymentPhase
+{
+    const Auth = 'AUTH';
```
Comment (sotuzun) [inline src/Model/PaymentPhase.php:7]: Uppercase yazmamız lazım. AUTH şeklinde.

Comment (AlicanAkkus) [reply src/Model/PaymentPhase.php]: done

File: src/Adapter/PaymentAdapter.php [hunk 2]
```diff
@@ -18,6 +18,12 @@ public function retrievePayment($paymentId)
         return $this->httpGet($path);
     }
 
+    public function postAuthPayment(array $request)
```
Comment (sotuzun) [inline src/Adapter/PaymentAdapter.php:21]: $paymentId, $request

Comment (sotuzun) [reply src/Adapter/PaymentAdapter.php:21]: public function searchWalletTransactions($walletId, array $request)
    {
        $path = "/wallet/v1/wallets/" . $walletId . "/wallet-transactions" . QueryBuilder::build($request);
        return $this->httpGet($path);
    }

Comment (AlicanAkkus) [reply src/Adapter/PaymentAdapter.php]: done

**Review by sotuzun — APPROVED** [review]

**Pull request merged**